tests/js/server/dump/dump-modified.js

```diff
@@ -277,6 +277,74 @@ function dumpTestSuite () {
       }
     },
 
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test padded keygen
+////////////////////////////////////////////////////////////////////////////////
+
+    testKeygenPadded : function () {
+      var c = db._collection("UnitTestsDumpKeygenPadded");
+      var p = c.properties();
+
+      assertEqual(2, c.type()); // document
+      assertFalse(p.waitForSync);
+      assertEqual("padded", p.keyOptions.type);
+      assertFalse(p.keyOptions.allowUserKeys);
+      assertEqual(process.env['paddedLastValue'], p.keyOptions.lastValue);
+
+      assertEqual(1, c.getIndexes().length); // just primary index
+      assertEqual("primary", c.getIndexes()[0].type);
+      assertEqual(1002, c.count());
+
+      let allDocs = {};
+      c.toArray().forEach(doc => {
+        allDocs[doc.value] = doc;
+      });
+      let lastKey = "";
+      for (var i = 0; i < 1000; ++i) {
+        var doc = allDocs[i];
+
+        assertTrue(doc._key > lastKey, doc._key + ">" + lastKey);
+        assertEqual(i, doc.value);
+        assertEqual({ value: [ i, i ] }, doc.more);
+        lastKey = doc._key;
+      }
+      doc = c.save({});
+      assertTrue(doc._key > lastKey, doc._key + ">" + lastKey);
+
+      // reset the value to before the one expected to be restored:
+      process.env['paddedLastValue'] = process.env['lastPaddedLastValue'];
+    },
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test uuid keygen
+////////////////////////////////////////////////////////////////////////////////
+
+    testKeygenUuid : function () {
```

Comment:
The same method I've just reviewed in "tests/js/server/dump/dump-mixed-cluster.js". 
Please re-use that method here. This is hard to maintain.